lib/modules/apostrophe-schemas/public/js/user.js

```diff
@@ -821,6 +829,52 @@ apos.define('apostrophe-schemas', {
         var nestedErrors = $fieldset.data('nestedErrors');
         $fieldset.data('nestedErrors', null);
         self.showError($fieldset, nestedErrors[0]);
+      },
+      contextualConvert: function(data, name, $el, field) {
```

Comment:
Where is this method called?

Reply:
When serializing a widget to be saved in the area editor, this method is invoked for any schema properties that have it, allowing nested areas/singletons to get saved too. It's present for arrays but was missing for objects.